LiteCore/Query/SQLiteN1QLFunctions.cc

```diff
@@ -813,28 +837,223 @@ namespace litecore {
 
 #pragma mark - DATES:
 
+    inline int64_t getQuarter(const DateTime& t) {
```

Comment:
This is a lot of code being added — could you modularize this by creating new source/header files for the date-time logic, and then having just the SQLite glue code in here?